enumerables.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each
+    length.times do |v|
+      yield(self[v])
+    end
+  end
+
+  def my_each_with_index
+    length.times do |v|
+      yield(self[v], v)
+    end
+  end
+
+  def my_select
+    arr = []
+    length.times do |v|
+      arr.push(self[v]) if yield(self[v])
+    end
+    arr
+  end
+
+  def my_all?
+    result = true
+    length.times do |v|
+      result = false if yield(self[v]) != true
+      break unless result == true
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    length.times do |v|
+      result = true if yield(self[v])
+      break unless result == false
+    end
+    result
+  end
+
+  def my_none?
+    result = true
+    length.times do |v|
+      result = false if yield(self[v])
+      break unless result == true
+    end
+    result
+  end
+
+  def my_count
+    result = 0
+    length.times do |v|
+      result += 1 if yield(self[v])
+    end
+    result
+  end
+
+  def my_map
+    result = []
+    length.times do |v|
+      yiel_var = yield(self[v])
+      result.push(yiel_var)
+    end
+    result
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an Enumerator if no block is given
```ruby
array = Array.new(100){rand(0...9)}
array.my_map == Enumerator #true
```